### DISCUSSION TITLE
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

### DISCUSSION THREAD

**Discussion by i-kj** [guardstrikelab/carla_apollo_bridge#130] [Q&A] [answered]
Hello all,
When I tried to run carla inside my docker container I am gettint the below error:

_[smit@in-dev-docker:/apollo/carla_apollo_bridge/scripts/carla_apollo_bridge/scri
pts]$ ./docker_run_carla.sh
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?_

I tried trouble shooting the issue. I did the following:
1. I checked the docker status outside the docker container in my host environment using: _**sudo systemctl status docker**_

_docker.service - Docker Application Container Engine
     Loaded: loaded (/lib/systemd/system/docker.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2023-08-17 13:30:22 CEST; 19min ago
TriggeredBy: ● docker.socket
       Docs: https://docs.docker.com_

and found the docker is active.

2. I checked if daemon is running or not using: _**ps aux | grep docker_**
_[smit@in-dev-docker:/apollo/carla_apollo_bridge/scripts/carla_apollo_bridge/scripts]$ ps aux | grep docker
root        1465  0.0  0.2 2353352 77612 ?       Ssl  13:30   0:01 /usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock
smit        4423  0.0  0.0 1550080 24568 pts/0   Sl+  13:42   0:00 docker exec -u smit -ti d6bc44b5f114 bash -c cd /apollo/carla_apollo_bridge/scripts && bash
smit        5179  0.0  0.0  11480   996 pts/2    S+   13:53   0:00 grep --color=auto docker_

so I understood that the daemon is running but I assumed I might have some accessibility issue. so I checked: _**groups**_

_[smit@in-dev-docker:/apollo/carla_apollo_bridge/scripts/carla_apollo_bridge/scripts]$ groups
smit sudo audio video_

I perfomed: _**sudo usermod -aG docker smit**_

I restarted my machine and retried to run docker_run_carla.sh again and got the same issue...

_smit@in-dev-docker:/apollo/carla_apollo_bridge/scripts/carla_apollo_bridge/scri
pts]$ ./docker_run_carla.sh
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?_

I am clueless on what to do next. I am ready to provide any additional information to help me get this done. Any help would be really appreciated. 

Thank you.

Please note that I cloned carla apollo bridge inside docker container. Can you please guide me in the right direction.

**Comment by XiaoFei9704** [comment]
Please try:
```
sudo usermod -aG docker $USER
```
And run  ./docker_run_carla.sh again

**Reply by i-kj** [reply]
Thank you for your reply.

_[smit@in-dev-docker:/apollo/carla_apollo_bridge/scripts/carla_apollo_bridge/scri
pts]$ sudo usermod -aG docker $USER
usermod: group 'docker' does not exist_

Please note that I cloned and trying to run _docker_run_carla.sh_ inside my docker container. Also when I tried cloning carla apollo bridge in my host machine, I could successfully launch docker_run_carla.sh. But I am not sure if that is correct. I am really beginner in this. Please help.

Also if I had ran docker_run_carla.sh on host machine and have CARLA running successfully, would the CARLA simulation be accessible on host machine? so I can continue with the CARLA Apollo Bridge setup within the Docker container.

**Reply by XiaoFei9704** [reply, marked as answer]
Before run docker_run_carla.sh on host machine, please edit [scripts](https://github.com/guardstrikelab/carla_apollo_bridge/tree/master/scripts)
/carla-compose.yml
Delete the following:
```
ports:
            - "2000:2000"
            - "2001:2001"
            - "2002:2002"
```
And then you can access carla by localhost:2000

**Comment by i-kj** [comment]
![Problem](https://github.com/guardstrikelab/carla_apollo_bridge/assets/141944156/34bacf24-3e9c-41d3-858f-8fb3aaf3e9a2)
Thanks it helped.

but I am facing a new problem wherein Carla 0.9.14 is lauching itself many times despite closing it since I installed apollo bridge. I restarted my PC but still the issue is existing. Can you please help?

_mit@smit-BUSINESSline:~$ ps aux | grep carla
smit       13352  0.0  0.0   4644   796 ?        Ss   11:51   0:00 /bin/sh ./CarlaUE4.sh -quality-level=Low -carla-rpc-port=2000
smit       13389  324 16.9 7724804 5499712 ?     Sl   11:51  15:29 /home/carla/CarlaUE4/Binaries/Linux/CarlaUE4-Linux-Shipping CarlaUE4 -quality-level=Low -carla-rpc-port=2000
smit       13487  0.0  0.0  10000   720 pts/2    S+   11:56   0:00 grep --color=auto carla_

even after killing the PID, it still launches.  Also i have carla 0.9.13 already installed but because of always active 0.9.14 ..i could launch it for some of my other works.

**Comment by XiaoFei9704** [comment]
See the last line of scripts/carla-compose.yml:
`        restart: always
`
If you do not want Carla to auto-restart, just delete it

**Reply by i-kj** [reply]
I deleted  _restart: always_ and saved but still CARLAUE4 is launching.  
    I really appreciate your help so far. Can you please help me on this as well?

_version: '3.7'
services:
    simulator:
        image: carlasim/carla:0.9.14
        logging:
          options:
            max-size: "10m"
            max-file: "3"
        privileged: true
        environment:
          SDL_VIDEODRIVER: x11
          DISPLAY: $DISPLAY
        ports:
            - "2000:2000"
            - "2001:2001"
            - "2002:2002"
        network_mode: "host"
        deploy:
          resources:
            reservations:
              devices:
                - driver: nvidia
                  count: 1
                  capabilities: ["gpu"]

        command: ./CarlaUE4.sh  -quality-level=Low -carla-rpc-port=2000
        volumes:
            - /tmp/.X11-unix:/tmp/.X11-unix:rw
            - /etc/localtime:/etc/localtime
            - /etc/timezone:/etc/timezone_
